ably/error_test.go

```diff
@@ -54,3 +59,83 @@ func TestIssue127ErrorResponse(t *testing.T) {
 	assert.NotNil(t, err)
 	assert.Contains(t, err.Error(), errMsg)
 }
+
+func TestIssue_154(t *testing.T) {
```

Comment:
This tests a number of things that have nothing to do with #154. We just want to check that a HTTP response with 400 status code turns into an ErrorInfo with 400 status code.

There are other tests that test retries and fallbacks.